browser/ui/views/download/brave_download_item_view.cc

```diff
@@ -236,18 +236,32 @@ void BraveDownloadItemView::SetMode(download::DownloadItemMode mode) {
   }
 }
 
+void BraveDownloadItemView::UpdateLabels() {
+  DownloadItemView::UpdateLabels();
+  // Update visibility to avoid artifacts to be shown from upstream
+  file_name_label_->SetVisible(
```

Comment:
How about we only set this to `false` and risk showing it when chromium means to hide it.

```
if (is_origin_url_visible_ && (GetMode() == download::DownloadItemMode::kNormal)) {
  file_name_label_->SetVisible(false);
}
```

Reply:
Also basing it on `GetMode() == download::DownloadItemMode::kNormal` seems consistent with chromium's label visibility choices. But it seems inconsistent with our origin label rendering which uses `IsShowingWarningDialog()` and that uses `has_warning_label(mode_)`. Maybe they match up to the inverse, but it seems we need them to _exactly and always_ match up. Perhaps we need to remove `IsShowingWarningDialog` and have the origin paint conditional be the same `is_origin_url_visible_ && (GetMode() == download::DownloadItemMode::kNormal)`. We could even make a function called `ShouldShowOriginUrl()` which performs that conditional, and use that function for both the file name label visibility and the origin visibility.

Reply:
iiirc, risk of depending on chromium tends to crash this on an equality check below